scripts/rddepman.ts

```diff
@@ -0,0 +1,210 @@
+
+type VersionComparison = {
+  name: string;
+  currentVersion: string | AlpineLimaISOVersion;
+  latestVersion: string | AlpineLimaISOVersion;
+};
+
+const dependencies: Dependency[] = [
+  new tools.KuberlrAndKubectl(),
+  new tools.Helm(),
+  new tools.DockerCLI(),
+  new tools.DockerBuildx(),
+  new tools.DockerCompose(),
+  new tools.DockerProvidedCredHelpers(),
+  new tools.Trivy(),
+  new tools.GuestAgent(),
+  new tools.Steve(),
+  new tools.RancherDashboard(),
+  new tools.ECRCredHelper(),
+  new LimaAndQemu(),
+  new AlpineLimaISO(),
+  new WSLDistro(),
+  new HostResolverHost(), // we only need one of HostResolverHost and HostResolverPeer
+  new MobyOpenAPISpec(),
+];
+
+function git(...args: string[]): number | null {
+  const name = 'Rancher Desktop Dependency Manager';
+  const email = 'donotuse@rancherdesktop.io';
+  const result = spawnSync('git', args, {
+    env: {
+      ...process.env,
+      GIT_AUTHOR_NAME:     name,
+      GIT_AUTHOR_EMAIL:    email,
+      GIT_COMMITTER_NAME:  name,
+      GIT_COMMITTER_EMAIL: email,
+    },
+  });
+
+  if (result.error) {
+    throw result.error;
+  }
+
+  return result.status;
+}
+
+function printable(version: string | AlpineLimaISOVersion): string {
+  return typeof version === 'string' ? version : version.isoVersion;
+}
+
+function getBranchName(name: string, currentVersion: string | AlpineLimaISOVersion, latestVersion: string | AlpineLimaISOVersion): string {
+  return `rddepman/${ name }/${ printable(currentVersion) }-to-${ printable(latestVersion) }`;
+}
+
+function getTitle(name: string, currentVersion: string | AlpineLimaISOVersion, latestVersion: string | AlpineLimaISOVersion): string {
+  return `rddepman: bump ${ name } from ${ printable(currentVersion) } to ${ printable(latestVersion) }`;
+}
+
+/**
+ * Compares the versions of two dependencies, as defined in DependencyVersions.
+ */
+function compareVersions(version1: string | AlpineLimaISOVersion, version2: string | AlpineLimaISOVersion): boolean {
+  if (typeof version1 === 'string' && typeof version2 === 'string') {
+    return version1 === version2;
+  } else if (typeof version1 !== 'string' && typeof version2 !== 'string') {
+    return version1.isoVersion === version2.isoVersion && version1.alpineVersion === version2.alpineVersion;
+  }
+  throw new Error(`Types of version1 (${ version1 }) and version2 (${ version2 }) differ.`);
+}
+
+async function createDependencyBumpPR(name: string, currentVersion: string | AlpineLimaISOVersion, latestVersion: string | AlpineLimaISOVersion): Promise<void> {
+  const title = getTitle(name, currentVersion, latestVersion);
+  const branchName = getBranchName(name, currentVersion, latestVersion);
+
+  console.log(`Creating PR "${ title }".`);
+  await getOctokit().rest.pulls.create({
+    owner: GITHUB_OWNER,
+    repo:  GITHUB_REPO,
+    title,
+    base:  MAIN_BRANCH,
+    head:  branchName,
+  });
+}
+
+type PRListFn = ReturnType<Octokit['rest']['search']['issuesAndPullRequests']>;
```

Comment:
```suggestion
type PRSearchFn = ReturnType<Octokit['rest']['search']['issuesAndPullRequests']>;
```
(It's no longer listing PRs.)